crates/ruff_linter/src/rules/flake8_use_pathlib/rules/os_path_abspath.rs

```diff
@@ -63,12 +73,45 @@ pub(crate) fn os_path_abspath(checker: &Checker, call: &ExprCall, segments: &[&s
     if segments != ["os", "path", "abspath"] {
         return;
     }
-    check_os_pathlib_single_arg_calls(
-        checker,
-        call,
-        "resolve()",
-        "path",
-        is_fix_os_path_abspath_enabled(checker.settings()),
-        OsPathAbspath,
-    );
+
+    if call.arguments.len() != 1 {
+        return;
+    }
+
+    let Some(arg) = call.arguments.find_argument_value("path", 0) else {
+        return;
+    };
+
+    let arg_code = checker.locator().slice(arg.range());
+    let range = call.range();
+
+    let mut diagnostic = checker.report_diagnostic(OsPathAbspath, call.func.range());
+
+    if has_unknown_keywords_or_starred_expr(&call.arguments, &["path"]) {
+        return;
+    }
+
+    if !is_fix_os_path_abspath_enabled(checker.settings()) {
+        return;
+    }
+
+    diagnostic.try_set_fix(|| {
+        let (import_edit, binding) = checker.importer().get_or_import_symbol(
+            &ImportRequest::import("pathlib", "Path"),
+            call.start(),
+            checker.semantic(),
+        )?;
+
+        let replacement = if is_pathlib_path_call(checker, arg) {
+            format!("{arg_code}.resolve()")
+        } else {
+            format!("{binding}({arg_code}).resolve()")
+        };
+
+        Ok(Fix::applicable_edits(
+            Edit::range_replacement(replacement, range),
+            [import_edit],
+            Applicability::Unsafe,
+        ))
```

Comment:
Since it's always unsafe, we can just use `Fix::unsafe_edits` instead of `applicable_edits`.


```suggestion
        Ok(Fix::unsafe_edits(
            Edit::range_replacement(replacement, range),
            [import_edit],
        ))
```